### PR TITLE
feat(tui): add search/filter to provider picker in setup wizard

### LINKED ISSUE #392: feat(tui): add search/filter to provider picker in setup wizard
## Problem

The provider picker in the setup wizard (`internal/tui/provider_wizard.go`) is a plain scrollable list with no search/filter capability. With 28+ providers now available, scrolling to find a specific provider is tedious.

The model picker (`internal/tui/picker.go`) already has search via `commandPicker.applyQuery()` — case-insensitive substring matching on model fields. The provider picker lacks the equivalent.

## Requested behavior

Add a search/filter input to the provider step of the setup wizard, similar to how the model picker works:

- Typing filters the provider list in real-time (case-insensitive substring match on provider name, ID, and aliases)
- Clear display showing active filter state
- Backspace to delete filter characters
- Arrow keys navigate filtered results

## Why it belongs in Zero

- 28+ providers now makes the scrollable list cumbersome
- Consistency with the model picker UX (which already has search)
- Improves first-run setup experience

## Alternatives or workarounds considered

- Users can scroll through the full list (current behavior)
- The list is sorted by catalog order, which helps, but doesn't solve the discoverability problem

## Implementation details

The model picker search in `internal/tui/picker.go:94` (`applyQuery()`) already implements case-insensitive substring filtering. The provider wizard's model step (`internal/tui/provider_wizard.go:1484`) also has `filteredModels()` with the same pattern. The provider step needs an equivalent `filteredProviders()` method and a `providerSearch` field.

The `fuzzySubsequenceGap()` function in `internal/tui/autocomplete.go:643` could also be reused if fuzzy matching is preferred over simple substring matching.

## Examples, screenshots, or prior discussion

- Provider wizard: `internal/tui/provider_wizard.go:1245` (`renderProviderStep`)
- Model picker search: `internal/tui/picker.go:94` (`applyQuery`)
- Provider wizard model search: `internal/tui/provider_wizard.go:1427` (`renderModelSearch`)

### PR DESCRIPTION
**Pull request by Ashwinhegde19**
## Summary
- Add a search input to the provider step of the setup wizard
- Filters providers in real-time via case-insensitive substring matching on name, ID, and aliases
- Consistent with the existing model picker search UX (`/model` command)

## Changes
- `internal/tui/provider_wizard.go`: 92 additions, 6 deletions

## What changed
- Added `providerSearch` field to `providerWizardState` struct
- Added `appendProviderSearch()`, `deleteProviderSearchRune()`, `filteredProviders()`, `providerMatchesQuery()` methods
- Updated `renderProviderStep()` to show search input and render filtered results
- Updated `currentProvider()` to use filtered list when on provider step
- Updated `move()` to navigate within filtered results
- Added key handling: printable chars → search, backspace → delete, Ctrl+U → clear
- Added paste support for provider search
- Search is cleared when advancing/retreating from provider step

## Motivation
With 28+ providers now in the catalog, scrolling through the full list is tedious. The model picker already has search — this brings the same UX to the provider picker.

## Testing
- `go build ./...` passes
- `go test ./internal/tui/...` passes
- `go vet ./internal/tui/...` passes

Fixes #392

## Summary by CodeRabbit

* **New Features**
  * Added provider search and filtering in the setup wizard.
  * The provider step now includes a live search field (matches provider ID, name, and aliases).

* **Bug Fixes**
  * Selection and navigation stay consistent with the filtered results.
  * When no providers match, you get clear “no matching providers” feedback, and selection won’t advance unexpectedly.

* **Chores**
  * Search input supports backspace/delete, paste, and quick clearing with `Ctrl+U` (resetting selection to the first match).

### CODE REVIEW

**Review by Vasanthdev2004 — CHANGES_REQUESTED** [review]
## Verdict: Request changes — the core is right, but there's one confirmed edge bug and no tests

Good feature, and the hard part is done correctly: `advance()` remaps the *filtered* selection index back to the *full-list* index before clearing the search, so selection is preserved across the filtered→full swap. I verified the happy path locally — filtering `anthropic` then pressing Enter advances with `anthropic` (remapped to full index 2), and the existing wizard tests still pass. gofmt/vet clean.

Two things to fix before merge.

### 🔴 Confirmed bug: Enter on an empty match advances with the wrong provider
Type a query that matches nothing → the list shows `no matching providers` → press **Enter**, and the wizard **advances off the provider step selecting `gitlawb-opengateway`** (the first full-list provider), not what the user intended. I reproduced it:

```
after Enter on empty match: step=Endpoint selectedProvider=0 provider="gitlawb-opengateway"
```

Root cause: in `advance()`'s `providerWizardStepProvider` case, when `currentProvider()` returns the empty descriptor (no match), the remap is skipped, `providerSearch` is cleared, and `currentProvider()` then falls back to full-list `[selectedProvider]` which the last keystroke reset to `0`. Enter isn't caught by the search key-switch, so it reaches `advance()`.

**Fix:** guard the empty case at the top of that `advance()` branch (right after the `oauthMode` return):
```go
if len(wizard.filteredProviders()) == 0 {
    return // nothing selectable; Enter is a no-op on an empty filter
}
```

### 🟡 No tests for the filter/index logic
The PR changes only `provider_wizard.go` — no tests. This is exactly the fragile index-mapping surface that PR #402 just had to refactor the wizard tests around, so it needs coverage. Please add table/unit tests for:
- filtering narrows the list (`filteredProviders()` / `providerMatchesQuery` on name, id, alias, case-insensitive),
- selecting a **filtered** provider advances to that exact provider (the remap — e.g. filter to a mid/late provider, Enter, assert `currentProvider().ID`),
- **empty match + Enter is a no-op** (guards the bug above — red-green),
- clearing the search (`Ctrl+U` / backspace to empty) restores the full list with the selection consistent.

These are pure `providerWizardState` unit tests — no full TUI run needed (see the probes I used above).

### Nit
`appendProviderSearch`/`deleteProviderSearchRune` reset `selectedProvider = 0` on every edit (good — avoids a stale index past a narrowed filter); just make sure the new tests lock that in so a future refactor can't regress it.

Once the empty-Enter guard and tests are in, this is a clean, welcome UX win. Thanks @Ashwinhegde19!

**Review by gnanam1990 — CHANGES_REQUESTED** [review]
The happy path is solid — typing a query + Enter selects the right provider, and existing picker navigation is intact (no off-by-one in the filtered→full remap). But there's a confirmed bug plus a coverage gap.

**1. Empty-filter + Enter mis-selects.** When the search matches zero providers, Enter isn't a no-op (`advance()`, `provider_wizard.go:469-486`): `currentProvider()` returns the empty descriptor, the remap is skipped, the search clears, and the wizard advances — configuring the *first full-list provider* (`gitlawb-opengateway`). This is the still-open CHANGES_REQUESTED from @Vasanthdev2004. A 2-line guard fixes it: `if len(wizard.filteredProviders()) == 0 { return }`.

**2. No tests.** The new filtered-index logic (remap-on-advance, empty-match Enter, `selectedProvider=0` reset on edit, backspace/Ctrl+U restore) is exactly the fragile mapping that needs table tests — there are none.

Minor: in OAuth mode the `d`/`D` device-login shortcut is checked before the search box, so you can't type `d` into the filter; and `retreat()` Provider→Method doesn't clear a stale query. Once the empty-filter guard + a couple of tests are in, this is good.

File: internal/tui/provider_wizard.go [hunk 1]
```diff
@@ -450,6 +463,7 @@ func (wizard *providerWizardState) advance() {
 		if wizard.oauthMode {
 			return
 		}
+		wizard.providerSearch = ""
 		wizard.refreshModels()
```
[the remark this answers is not in the document]
Comment (Ashwinhegde19) [reply internal/tui/provider_wizard.go:481]: Already handled — lines 469-475 re-map  from the filtered index to the full list index before clearing the search. At line 480,  uses the updated index against the full list, so the correct provider is preserved.

File: internal/tui/provider_wizard.go [hunk 2]
```diff
@@ -1265,6 +1346,11 @@ func (wizard *providerWizardState) renderProviderStep(width int) []string {
 	return lines
 }
 
+func (wizard *providerWizardState) renderProviderSearch(width int) string {
+	query := strings.TrimSpace(wizard.providerSearch)
+	return providerWizardInputLine("search > ", query, "provider name...", width)
+}
```
[the remark this answers is not in the document]
Comment (Ashwinhegde19) [reply internal/tui/provider_wizard.go:1366]: Already updated — the placeholder now reads `"provider name, id, or alias..."` to reflect that search matches all three.

**Comment by Ashwinhegde19** [comment]
Addressed both items:

1. **Empty-filter + Enter bug**: Added guard `if len(wizard.filteredProviders()) == 0 { return }` at the top of the `providerWizardStepProvider` case in `advance()`. Enter is now a no-op when the search matches nothing.

2. **Tests added** (`provider_wizard_test.go`):
- `TestProviderSearchFiltersByNameIDAndAlias` — table-driven test covering name, ID, alias, case-insensitive, no-match, and empty query
- `TestProviderSearchResetsSelectedProviderOnEdit` — confirms `selectedProvider` resets to 0 on typing
- `TestProviderSearchFilteredAdvanceSelectsCorrectProvider` — verifies `currentProvider()` returns the right provider from the filtered list
- `TestProviderSearchEmptyMatchEnterIsNoop` — guards the empty-filter bug (red-green)
- `TestProviderSearchClearRestoresFullList` — Ctrl+U clears search and restores full provider list

All tests pass, `go vet` clean. PTAL @Vasanthdev2004 @gnanam1990

**Review by gnanam1990 — APPROVED** [review]
## VERDICT: approve

## REGRESSION RISK: low

Re-review of new commits since my prior review (`236195b5` → `a44d96c1`). The new commits add a guard for Enter key when the filter is empty (preventing unintended selection) and add 136 lines of provider search tests. The guard is a 3-line addition in `provider_wizard.go` that prevents Enter from acting on an empty filter state. No existing behavior is altered for non-empty filters.

## BUILD / TEST

- CI all green (macOS, Ubuntu, Windows, Smoke, Security, Zero Review)
- The new tests cover the search/filter behavior comprehensively

## FINDINGS

None. The empty-filter guard is a reasonable edge-case fix and the test coverage is thorough.

## BOTTOM LINE

The empty-filter Enter guard and expanded test coverage address the prior review's edge-case concern. Safe to merge.

**Review by kevincodex1 — APPROVED** [review]
lgtm
